crypto/fipsmodule/bcm.c

```diff
@@ -386,7 +376,13 @@ int BORINGSSL_integrity_test(void) {
 }
 #endif  // OPENSSL_ASAN
 
-void BORINGSSL_FIPS_abort(void) {
+WEAK_SYMBOL_FUNC(void, AWS_LC_fips_failure_callback, (const char* message))
+#include <unistd.h>
+void AWS_LC_FIPS_failure(const char* message) {
```

Comment:
Please correct me if I'm wrong, but this will break our next FIPS validation.  If we allow someone outside the boundary to register a callback our lab will not be able to prove that we stop processing events when a PCT or self-tests fails. When we detect a failure we need to abort immediately as defined in ISO 19790 section 7.3.3, b), 7.3.3 c), and 7.5. This needs to be wrapped in a #ifdef so we don't expose it when we are building as a 140-3 level 1 module.

Reply:
This is no different than the current behavior: an application using AWS-LC can install a custom signal handler to perform any logging/cleanup before the abort terminates the process. However, with this callback the application gets a clearer signal if the abort is a FIPS anomaly or a different SIGABRT.

Reply:
Updated to a callback that doesn't abort if set and put it all behind a compilation option.